src/facet/data/partition/_partition.py

```diff
@@ -111,7 +115,9 @@ def is_categorical(self) -> bool:
         """
 
     @abstractmethod
-    def fit(self: T_Self, values: Iterable[T_Values], **fit_params: Any) -> T_Self:
+    def fit(
+        self: T_Partitioner, values: Sequence[T_Values], **fit_params: Any
```

Comment:
I think it must be a sequence because in e.g. class `RangePartitioner` that inherits from `Partitioner` in `fit` implementation access by index is used `values[values >= q3q1[1] - inlier_range]`. `Iterable` does not support `__getitem__()` where `Sequence` does. Thus to mypy typecheck correctly I specialized that variable.